_drafts/partially-ordering-time.md

```diff
@@ -0,0 +1,253 @@
+as mine? If time wasn't weird, if we didn't constantly take license to reorder operations
+in all of these systems, the answer would be yes, of course.
+
+But time is weird, and so we have to ask.
+
+### Consistency Models—I Mean, Memory Models
+
+Both these worlds started out by trying to ignore time being this weird for as long as
+they could—distributed programmers tried to hue to ACID properties as long as possible,
+and systems programmers relied on mutual exclusion / locks for a long time.
+
+But we all need speed. Whether it's the CAP theorem and weakly/eventually consistent
+models [#url], or lockfree programming with weak/relaxed memory models, both worlds ended
+up relaxing this strong view of time in order to gain speed.
+
+Weak memory models and consistency models are both _abstract_ models: they describe the
+_interface_ of the system to the programmer using it. They describe what sorts of
+behaviours we _can_ rely on from time, now that the default properties we tend to assume
+no longer apply.
+
+Both worlds have developed their own language for describing these models. For instance:
+the phrase “data race” in systems programming refers to any pair of accesses to a memory
+location that include at least one write, without a synchronisation barrier between
+them. This concept doesn't have a specific phrasing in distributed programming, because
+that field assumes that reads and writes participate in data races (“are concurrent”) by
+default.
+
+Thus, “data-race freedom” is a stronger condition than the distributed systems term
+“sequential consistency” (“any concurrent accesses can be treated _as if_ they were
+synchronised”), but if you can prove that a program on your processor is data-race free,
+you can _provide_ sequential consistency to that program as your hardware or software
+memory model (“SQ-DRF”, “sequential consistency for data-race free programs” [6]).
+
+…You can see how this could get confusing.
+
+The obvious solution is, of course, to make things more confusing!
+
+### The Burckhardt Paper
+
+The Burckhardt Paper. The _Burckhardt paper_.
+
+Dr. Sebastian Burckhardt has [been trying](#urls) to formalise the specification of
+consistency models for some time, in the distributed systems world. In [7], Appendix D.4,
+their team lays out an analysis of the C++ weak memory model in a slight specialisation of their framework.
+
+This gives us a convenient way to discuss the similarities and differences between the two
+fields. In particular, the C++ memory model _approximates_ to per-object causal
+consistency, with a few interesting differences.
+
+We'll study them axiom by axiom, but it's first important to note that where causal
+consistency is defined in (Burckhardt's) terms of a “visibility” and an “arbitration”
+orderings, they define weak memory axioms in terms of a “reads-from” and “modification
+order” instead, correspondingly.
```

Comment:
Hm, I'm not sure if this is a regional thing, but I think "respectively" should be used instead of "correspondingly"

I also think you can omit "We'll study them axiom by axiom" in this sentence. If you'd like, you can just move it into the previous paragraph or something.

I also think they doesn't have a clear modifier. I'm not sure if "reads-from" and "modification order" are part of the specialized model from Burckhardt, or from the weak memory model.